miniapps/nurbs/nurbs_naca_cmesh.cpp

```diff
@@ -0,0 +1,792 @@
+
+#include "mfem.hpp"
+#include <iostream>
+#include <cmath>
+
+using namespace std;
+using namespace mfem;
+
+// Object that discribes a symmetric NACA foil section
+class NACA4
+{
+protected:
+   // Constants describing the thickness profile
+   double A, B, C, D, E;
+   // Thickness of the foil section
+   double t;
+   // Chord of the foil section: foil length
+   double c;
+   // Maximum number of iterations for Newton solver
+   int iter_max;
+   // Tolerance for Newton solver
+   double epsilon;
+public:
+   NACA4(double t_, double c_);
+   // Returns the coordinate y corresponding to coordinate @a xi
+   double y(double xi);
+   // Returns the derivative of the curve at location coordinate @a xi
+   double dydx(double xi);
+   // Returns the curve length at coordinate @a xi
+   double len(double xi);
+   // Returns the derivative of the curve length at coordinate @a xi
+   double dlendx(double xi);
+   // Returns the coordinate x corresponding to the curve length @a l from
+   // the tip of the foil section
+   double xl(double l);
+   // Get the chord of the foil_section
+   const double GetChord() {return c;}
+};
+
+// Function that finds the coordinates of the control points of the tip of the foil section @a xy
+// based on the @a foil_section, knot vector @a kv and tip fraction @a tf
+// We have two cases, with an odd number of control points and with an even number of
+// control points. These may be streamlined in the future.
+void GetTipXY(NACA4 foil_section, KnotVector *kv, double tf,
+              Array<Vector*> &xy);
+
+// Function that returns a uniform knot vector based on the @a order and the
+// number of control points @a ncp.
+KnotVector *UniformKnotVector(int order, int ncp);
+
+// Function that returns a knot vector which is stretched with stretch @s
+// with the form x^s based on the @a order and the number of control points @a ncp.
+// Special case @a s = 0 will give a uniform knot vector.
+KnotVector *PowerStretchKnotVector(int order, int ncp, double s = 0.0);
+
+// Function that returns a knot vector with a hyperbolic tangent spacing
+// with a cut-off @c using the @a order and the number of control points @a ncp.
+KnotVector *TanhKnotVector(int order, int ncp, double c);
+
+// Function that returns a knot vector with a hyperbolic tangent spacing from both sides
+// of the knot vector with a cut-off @c using the @a order and the number of control points @a ncp.
+KnotVector *DoubleTanhKnotVector(int order, int ncp, double c);
+
+// Function that evaluates a linear function which describes the boundary distance
+// based on the flair angle @a flair, smallest boundary bistance @a bd and
+// coordinate @a x. The flair angle is mainly used to be able to enforce inflow on
+// the top and bottom boundary and to create an elegant mesh.
+double FlairBoundDist(double flair, double bd, double x);
+
+int main(int argc, char *argv[])
+{
+   int mdim = 2;
+   int order = 2;
+
+   //
```

Comment:
These empty comment lines are not usually used in the MFEM examples.

Reply:
Removed